server/routes/auth.ts

```diff
@@ -1,9 +1,34 @@
-import express from "express";
-import pages from "../config/auth-pages.json";
+import express, { Request, Response } from "express";
+import customAdminPages from "../config/admin/admin-pages.json";
+import { BackendConstructor } from "./backend-constructor";
 import { PageLoader } from "./page-loader";
 
 const router = express.Router();
 
-PageLoader.buildRoutes(pages, router);
+const pageBuilder = (pageData) => {
+  pageData["adminPages"] = customAdminPages;
+  return pageData;
+};
+
+PageLoader.buildRoutes(customAdminPages, router, pageBuilder);
+
+router.get("/admin/editor", async (req: Request, res: Response) => {
+  const file = req.query["page"];
+
+  const pipedData = PageLoader.getPagePipedData(
+    {
+      title: "Editor",
+      ref: "/admin/editor",
+    },
+    pageBuilder
+  );
+
+  const dataPages = BackendConstructor.getDataDirectoryStructure();
+
+  pipedData["editors"] = dataPages;
```

Comment:
Since these are only used for the /generic-editor page, I think adding them to any interface might be a bit overkill.  I've updated the type safety though as inspired by your comments.